_modules/graph-visualiser/main.tf

```diff
@@ -3,15 +3,20 @@ locals {
     [for i in var.authorised_users.groups : "group:${i}"],
     [for i in var.authorised_users.users : "user:${i}"]
   ))
+
+  # projects/{{project}}/regions/{{region}}/subnetworks/{{name}}
+  subnet_parts   = split("/", var.subnetwork)
+  subnet_project = local.subnet_parts[1]
+  subnet_name    = local.subnet_parts[5]
 }
 
 data "google_project" "data_plane" {
   project_id = var.project_id
 }
 
 data "google_compute_subnetwork" "subnetwork" {
-  project = var.project_id
-  name    = var.subnetwork
+  project = local.subnet_project
+  name    = local.subnet_name
```

Comment:
Just wondering whether possible to pass straight self_link vs splitting and explicitly setting separate args
https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/compute_subnetwork#argument-reference